small_llm_pretraining/nemo/README.md

```diff
@@ -0,0 +1,226 @@
+- Training partitions: `c4-train.en_<number>_text_document`
+- Validation partitions: `c4-validation-91205-samples.en_text_document`
+
+#### Training and test data separation
+
+We use the default split from the C4 dataset. This means that we use `c4-train.<x>-of-01024.json.gz` files (where `768 <= x <= 1023`) for training, and we use our customized `c4-validation-91205-samples.en.json.gz`, which contains the first 91205 samples from the unshuffled C4 validation dataset, for evaluation. 
+
+Notice here that we are using the first 1024 sequences (8,388,608 tokens) from the validation dataset to perform the validation. According to our experiments, the first 91205 samples from the unshuffled C4 dataset yields 47,186,855 tokens, which is the smallest amount of samples needed to yield 47,185,920 tokens. Thus, we have chosen the first 91205 samples as our validation dataset. 
+
+#### Training data order
+
+We randomly shuffle the **last 256 of 1024 shards** for the benchmarking area.
+
+#### Test data order
+
+We use the first 1024 sequences in the validation dataset for validation. We **do not shuffle** the validation dataset. 
+
+# 4. Model
+### Publication/Attribution
+
+The model largely follows the Llama 3.1 8B [paper](https://arxiv.org/abs/2407.21783). 
+
+### Model details
+
+| Config | Value |
+| :-- | :-- | 
+| Embedding | RoPE + parameter adjustments |
+| # Layers | 32 | 
+| Attention Type | GQA |
+| # Attn Heads | 32 | 
+| Key/Value Heads | 8 | 
+| Model Dimension | 4096 |
+| FFN Dimension | 14336 |
+| Activation | SwiGLU | 
+| Normalization | RMSNorm |  
+| Tokenizer | Llama tokenizer |
+| Vocab size | 128,000 |  
+| Context Length | 8192 |
+
+
+#### Saving and restoring a checkpoint
+
+Large runs might need to span across multiple Slurm jobs, and we need to save and load checkpoints with contexts so that training can resume between jobs. To support this, we have added some environment variables. Please refer to `config.sh` for more details. 
+
+### Optimizer spec
+
+1. Optimizer type: **AdamW**
+2. Warmup steps computed as 10% of the total allocated steps.
+
+# 5. Quality
+### Quality metric
+
+Validation loss
+
+### Quality target
+
+Validation log perplexity = 3.3
+
+### Evaluation frequency
+
+We perform evaluation every **12288** sequences. 
+
+### Evaluation thoroughness
+
+We evaluate using **1024** sequences from our customized validation dataset. 
+
+# 6. Launch a training run
+
+To train Llama 3.1 8B, we need to fill out all fields in `config.sh`. This file contains all configurations for Slurm cluster access and job submission configurations, directory mappings, containers, and model configurations. 
+
+Once the `config.sh` is properly filled, we launch a training run using the following commands:
+
+```bash
+source config.sh
+bash run_llama31.sh
+```
+
+<!-- # 6. Other
+
+#### Run model conversion
```

Comment:
Can we drop this section because there is no starting checkpoint for this benchmark. Submitters **should not** start from the HF checkpoint and instead they need to start from randomly initialized weights